frontend/src/components/ui/WelcomeModal.jsx

```diff
@@ -1,15 +1,38 @@
 import Button from 'react-bootstrap/Button';
 import Modal from 'react-bootstrap/Modal';
+import Game from '../../Game';
+import { useState } from 'react';
+
+function WelcomeModal({ show, hideModal, setGame }) {
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const loadGame = async (e) => {
+    e.preventDefault();
+    try {
+      const clipboardData = await navigator.clipboard.readText();
+
+      const game = Game.from(clipboardData);
+      if (game.error) {
+        setErrorMessage(game.error.message);
+      } else {
+        setGame(game);
+        hideModal();
+      }
+
+    } catch (err) {
+      console.error('Failed to load game:', err);
+      setErrorMessage('Failed to load game from clipboard.' + err.message);
+    }
+  }
 
-function WelcomeModal(props) {
   return (
     <Modal
-      {...props}
+      {...{ show }}
```

Comment:
Any reason to keep the spread operator here?
```suggestion
      show
```